parsl/dataflow/dflow.py

```diff
@@ -1115,7 +1115,7 @@ def _create_remote_dirs_over_channel(self, provider: ExecutionProvider, channel:
 
         channel.makedirs(channel.script_dir, exist_ok=True)
 
-    def add_executors(self, executors):
+    def add_executors(self, executors: Sequence[ParslExecutor]) -> None:
```

Comment:
I'll make a separate PR for this. I think the `executors` parameter can already be typed as `Iterable` rather than `Sequence`? it's only used in an iteration context.

Reply:
wups, i read your diff the wrong way round - I prefer the style it is now: executors is a small structure usually, and I'd prefer to not introduce more mutable state in this function vs having a single line that shows what the filter on block_executors is.

Reply:
turning job_status_poller.add_executors into a single-executor method that gets called in every iteration, without building the block_executors iterable at all, might be an ok thing to do here, though... it's not an outwards facing API. but there's a whole chain of tidyups that could happen in that callstack.